### PR TITLE
Fix csv examples

### LINKED ISSUE #3174: Fix CSV docs
Related PR: https://github.com/influxdata/flux/issues/4054

Review, update, and test the CSV examples in the following docs to be sure they're valid and accurate:
- https://docs.influxdata.com/influxdb/cloud/write-data/developer-tools/csv/: 
   - Clarify that Flux needs all of the following three annotations to work: `datatype,group,default`
   - Global search to add a comma between datatype and measurement here: `#datatype measurement,tag,double,dateTime:RFC3339`
- https://docs.influxdata.com/influxdb/v2.0/reference/syntax/annotated-csv/ : 
   - Clarify the `to()` function needs to know which column is the measurement (by default, `to()` assumes the `_measurement` column). Add  `_measurement` column to the example CSV data. 
   - Add a note to distinguish the different supported methods and variances--while you can leave the measurement out of the `influx write` command, if you leave the measurement out of data to upload csv or leave measurement out of the Flux `csv.from` function, it's going to error.  Is leaving the "measurement" out the only distinction to be made here, @nathanielc ?

cc @barbaranelson @nathanielc and @genehynson for review

### PR DESCRIPTION
**Pull request by mcfarlm3**
Closes #3174

This PR attempts to address CSV examples which resulted in errors due to incorrect syntax and formatting differences between `influx write` and Flux. My observations after testing the examples, and corresponding changes are listed below.

Observations:
- `influx write` requires a space between the annotation name and annotation values (will error if a comma is added instead), while Flux requires a comma between annotation names and annotation values.
- Flux requires `datatype`, `group`, and `default` annotations. If any of these 3 are missing, it will produce an error.
- Flux requires the `_measurement` column to be present
- The `--skip-headers` flag for `influx write` used to be supported as a bool and int (making `--skip-headers` equivalent to `--skip-headers=1`). Now, this flag is only treated as an integer, meaning that `--skip-headers` is now equivalent to `--skip-headers=0`.

Changes made:
- Fixed Flux CSV example [here](https://docs.influxdata.com/influxdb/v2.0/reference/syntax/annotated-csv/#annotated-csv-in-flux). Now it includes all required annotations and the `_measurement` column. 
- Changed [this](https://docs.influxdata.com/influxdb/v2.0/write-data/developer-tools/csv/#replace-column-header-with-annotation-shorthand) example to reflect the change in the`--skip-headers` flag (default of 0 instead of 1)
-  Added additional information to [this](https://docs.influxdata.com/influxdb/v2.0/write-data/developer-tools/csv/#required-columns) note to try to differentiate what is needed for Flux vs. `influx write` CSV data

The differences between requirements for `influx write` and Flux may still be too ambiguous - if there's more I should add to make those differences clear, feedback would be great.

- [x] Signed the [InfluxData CLA](https://www.influxdata.com/legal/cla/)
  ([if necessary](https://github.com/influxdata/docs-v2/blob/master/CONTRIBUTING.md#sign-the-influxdata-cla))
- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

File: content/influxdb/v2.0/reference/syntax/annotated-csv/_index.md [hunk 1]
```diff
@@ -252,16 +252,16 @@ to read annotated CSV in Flux:
...
-#default,,,,,,,,
-,result,table,_start,_stop,_time,region,host,_value
-,,0,2018-05-08T20:50:00Z,2018-05-08T20:51:00Z,2018-05-08T20:50:00Z,east,A,15.43
-,,0,2018-05-08T20:50:00Z,2018-05-08T20:51:00Z,2018-05-08T20:50:20Z,east,B,59.25
-,,0,2018-05-08T20:50:00Z,2018-05-08T20:51:00Z,2018-05-08T20:50:40Z,east,C,52.62
-,,1,2018-05-08T20:50:00Z,2018-05-08T20:51:00Z,2018-05-08T20:50:00Z,west,A,62.73
-,,1,2018-05-08T20:50:00Z,2018-05-08T20:51:00Z,2018-05-08T20:50:20Z,west,B,12.83
-,,1,2018-05-08T20:50:00Z,2018-05-08T20:51:00Z,2018-05-08T20:50:40Z,west,C,51.62
+csvData = "#datatype,string,long,dateTime:RFC3339,dateTime:RFC3339,dateTime:RFC3339,string,string,double,string
+#group,false,false,false,false,false,false,false,false,false
```
Comment (sanderson) [inline content/influxdb/v2.0/reference/syntax/annotated-csv/_index.md:256]: The group annotation doesn't accurately reflect the grouping of the csv data.

```suggestion
#group,false,false,true,true,false,true,false,false,true
```

File: content/influxdb/v2.0/write-data/developer-tools/csv.md [hunk 2]
```diff
@@ -586,8 +581,19 @@ csv.from(url: "https://influx-testdata.s3.amazonaws.com/noaa.csv")
 ```
 
 {{% note %}}
-#### Required columns
-To write data to InfluxDB with Flux, data must include the following columns:
+#### Required annotations and columns
+To write CSV data to InfluxDB with Flux, ALL of the following annotations and columns must be present.
```
Comment (sanderson) [inline content/influxdb/v2.0/write-data/developer-tools/csv.md:585]:
```suggestion
To write CSV data to InfluxDB with Flux, _all_ of the following annotations and columns must be present.
```

File: content/influxdb/v2.0/write-data/developer-tools/csv.md [hunk 3]
```diff
@@ -586,8 +581,19 @@ csv.from(url: "https://influx-testdata.s3.amazonaws.com/noaa.csv")
...
+Required annotations:
+
+- `datatype`
+- `group`
+- `default`
+
+See [annotations](/influxdb/cloud/reference/syntax/annotated-csv/#annotations) for more information.
```
Comment (sanderson) [inline content/influxdb/v2.0/write-data/developer-tools/csv.md:592]:
```suggestion
See [annotations](/influxdb/v2.0/reference/syntax/annotated-csv/#annotations) for more information.
```

**Review by sanderson — CHANGES_REQUESTED** [review]
Thanks for this @mcfarlm3. Could you also port these changes into the new InfluxDB 2.1 docs.

**Comment by mcfarlm3** [comment]
> Thanks for this @mcfarlm3. Could you also port these changes into the new InfluxDB 2.1 docs.

Definitely - I can get that done today or tomorrow at the latest. Thanks for the feedback!

**Review by sanderson — APPROVED** [review]
Thanks @mcfarlm3!

File: content/influxdb/v2.1/write-data/developer-tools/csv.md [hunk 4]
```diff
@@ -586,8 +581,19 @@ csv.from(url: "https://influx-testdata.s3.amazonaws.com/noaa.csv")
 ```
 
 {{% note %}}
-#### Required columns
-To write data to InfluxDB with Flux, data must include the following columns:
+#### Required annotations and columns
+To write CSV data to InfluxDB with Flux, _all_ of the following annotations and columns must be present.
+Required annotations:
```
Comment (kelseiv) [inline content/influxdb/v2.1/write-data/developer-tools/csv.md:586]:
```suggestion
To write CSV data to InfluxDB with Flux, you must include _all_ of the following annotations and columns:
```

File: content/influxdb/v2.1/write-data/developer-tools/csv.md [hunk 5]
```diff
@@ -586,8 +581,19 @@ csv.from(url: "https://influx-testdata.s3.amazonaws.com/noaa.csv")
...
+
+- `datatype`
+- `group`
+- `default`
+
+See [annotations](/influxdb/v2.1/reference/syntax/annotated-csv/#annotations) for more information.
+With Flux, there must also be a comma between the annotation name and the annotation values (this differs from the `influx write` command).
```
Comment (kelseiv) [inline content/influxdb/v2.1/write-data/developer-tools/csv.md:593]:
```suggestion
With Flux, you must also include a comma between the annotation name and the annotation values (this differs from the `influx write` command).
```

File: content/influxdb/v2.1/write-data/developer-tools/csv.md [hunk 6]
```diff
@@ -586,8 +581,19 @@ csv.from(url: "https://influx-testdata.s3.amazonaws.com/noaa.csv")
...
+An example of valid Flux syntax can be found [here](/influxdb/v2.1/reference/syntax/annotated-csv/#annotated-csv-in-flux).
```
Comment (kelseiv) [inline content/influxdb/v2.1/write-data/developer-tools/csv.md:594]:
```suggestion
See an example of valid syntax for [annotated CSV in Flux](/influxdb/v2.1/reference/syntax/annotated-csv/#annotated-csv-in-flux).
```

File: content/influxdb/v2.0/write-data/developer-tools/csv.md [hunk 7]
```diff
@@ -586,8 +581,19 @@ csv.from(url: "https://influx-testdata.s3.amazonaws.com/noaa.csv")
 ```
 
 {{% note %}}
-#### Required columns
-To write data to InfluxDB with Flux, data must include the following columns:
+#### Required annotations and columns
+To write CSV data to InfluxDB with Flux, _all_ of the following annotations and columns must be present.
+Required annotations:
```
Comment (kelseiv) [inline content/influxdb/v2.0/write-data/developer-tools/csv.md:586]:
```suggestion
To write CSV data to InfluxDB with Flux, you must include _all_ of the following annotations and columns:
```

File: content/influxdb/v2.0/write-data/developer-tools/csv.md [hunk 8]
```diff
@@ -586,8 +581,19 @@ csv.from(url: "https://influx-testdata.s3.amazonaws.com/noaa.csv")
...
+
+- `datatype`
+- `group`
+- `default`
+
+See [annotations](/influxdb/v2.0/reference/syntax/annotated-csv/#annotations) for more information.
+With Flux, there must also be a comma between the annotation name and the annotation values (this differs from the `influx write` command).
```
Comment (kelseiv) [inline content/influxdb/v2.0/write-data/developer-tools/csv.md:593]: With Flux, you must also include a comma between the annotation name and the annotation values (this differs from the `influx write` command).

File: content/influxdb/v2.0/write-data/developer-tools/csv.md [hunk 9]
```diff
@@ -586,8 +581,19 @@ csv.from(url: "https://influx-testdata.s3.amazonaws.com/noaa.csv")
...
+An example of valid Flux syntax can be found [here](/influxdb/cloud/reference/syntax/annotated-csv/#annotated-csv-in-flux).
```
Comment (kelseiv) [inline content/influxdb/v2.0/write-data/developer-tools/csv.md:594]:
```suggestion
See an example of valid syntax for [annotated CSV in Flux](/influxdb/v2.1/reference/syntax/annotated-csv/#annotated-csv-in-flux).
```

**Review by kelseiv — APPROVED** [review]

**Pull request merged**